### DISCUSSION TITLE
关于PartitionStatistics和直接query对应partition的数量差异问题

### DISCUSSION THREAD

**Discussion by 131452099900** [milvus-io/milvus#31604] [Q&A and General discussion] [answered]
我这边使用的是java的sdk，在使用client的getPartitionStatistics方法时也就是统计某个partition的实体数量，比如我在这边查到有100条的数据；然后我用另一种通过query查询partition得到的数据有98条，比上面的少了两条，有这么一种数据的差异。
不过我测了几个大部分情况上面两种是相等的，但是有一个是比较异常的，上面这种请求，请问第一种是直接根据我们etcd的元数据直接获取的么，而query查到的应该是真实的数量吧。

**Comment by yhmo** [comment, marked as answer]
在Java SDK里，getPartitionStatistics/getCollectionStatistics 就相当于 Python SDK中的 collection.num_entities 方法，这种方法是从etcd中拿取已落盘的segment的行数，它不需要预先调用 collection.load()，速度快，但是不准。

而 query(expr="", output_fields=["count(*)"]) 则需要预先调用 collection.load()，它可以统计未落盘的行数，所以比较准确，速度慢。
